zk-token-sdk/src/zk_token_proof_instruction.rs

```diff
@@ -6,11 +6,21 @@
 //! Each proof verification instruction verifies a certain type of zero-knowledge proof. These
 //! instructions are processed by the program in two steps:
 //!   1. The program verifies the zero-knowledge proof.
-//!   2. The program optionally stores the context component of the instruction data to a
+//!   2. The program optionally stores the context component of the zero-knowledge proof to a
 //!      dedicated [`context-state`] account.
-//! If no accounts are provided with the instruction, the program simply verifies the proofs. If
-//! accounts are provided with the instruction, then the program writes the context data to the
-//! specified context-state account.
+//!
+//! In step 1, the zero-knowledge proof can be included directly as the instruction data or
+//! pre-written to an account. The program determines whether the proof is provided as instruction
+//! data or pre-written to an account by inspecting the length of the data. If the instruction data
+//! is exactly 4 bytes, then the program assumes that the first account provided with the
```

Comment:
nit, does this make more sense?
```suggestion
//! is exactly 5 bytes, then the program assumes that the first account provided with the
```

Reply:
Yeah, I think 4 bytes makes things more confusing. I updated to 5 bytes and then in parenthesis specified (discriminator + offset).